include/ginkgo/core/base/math.hpp

```diff
@@ -477,6 +430,26 @@ using next_precision = typename detail::next_precision_impl<T>::type;
 template <typename T>
 using previous_precision = next_precision<T>;
 
+/**
+ * Obtains the next type in the singly-linked precision list with half.
+ */
+#if GINKGO_ENABLE_HALF
+template <typename T>
+using next_precision_with_half =
```

Comment:
The old `next_precision` is still used in the combination, composition, and pertubation reference test.